src/realm/cluster.cpp

```diff
@@ -1944,78 +1944,80 @@ Obj ClusterTree::insert(ObjKey k, const FieldValues& values)
 
     insert_fast(k, init_values, state);
 
-    // Update index
-    auto value = init_values.begin();
-    auto insert_in_column = [&](ColKey col_key) {
-        // Check if initial value is provided
-        Mixed init_value;
-        if (value != init_values.end() && value->col_key.get_index().val == col_key.get_index().val) {
-            init_value = value->value;
-            ++value;
-        }
+    if (!k.is_unresolved()) {
```

Comment:
Ìs this purely an optimization or does it fix an error?

Reply:
Solves a problem